app/controllers/correctReturn/CorrectReturnCheckChangesCYAController.scala

```diff
@@ -69,8 +74,16 @@ class CorrectReturnCheckChangesCYAController @Inject()(
       }).getOrElse(Future.successful(Redirect(controllers.routes.SelectChangeController.onPageLoad.url)))
   }
 
-  def onSubmit: Action[AnyContent] = controllerActions.withRequiredJourneyData(CorrectReturn) {
-    Redirect(routes.CorrectReturnUpdateDoneController.onPageLoad.url)
+  def onSubmit: Action[AnyContent] = controllerActions.withRequiredJourneyData(CorrectReturn).async {
+    implicit request =>
+    correctReturnOrchestrator.submitVariation(request.userAnswers, request.subscription).map(result =>
+      result.value.map {
+        case Right(_) =>  Redirect(routes.CorrectReturnUpdateDoneController.onPageLoad.url)
+        case Left(_) => genericLogger.logger.error(s"${getClass.getName} - ${request.userAnswers.id} - failed to submit return variation")
+          InternalServerError(errorHandler.internalServerErrorTemplate)
+      }).getOrElse{
+      genericLogger.logger.error(s"${getClass.getName} - ${request.userAnswers.id} - failed to submit return variation due to retrieving information")
```

Comment:
I would suggest a more clear message about what is failing, such as "failed to submit return variation due to failure when retrieving information from server")